### PR TITLE
ocm-addons-upgrade-scheduler-org integration

### PR DESCRIPTION
**Pull request by patjlm**
https://issues.redhat.com/browse/APPSRE-6790

proposal to manage addon upgrade based on our existing upgrade policies. This is targeting hypershift only, as a short term solution

Depends on https://github.com/app-sre/qontract-schemas/pull/368

### CODE REVIEW

File: reconcile/utils/ocm.py [hunk 1]
```diff
@@ -1194,7 +1202,8 @@ def version_blocked(self, version):
         Returns:
             bool: is version blocked
         """
-        return any(re.search(b, version) for b in self.blocked_versions)
+        s = f"{addon_id}/{version}" if addon_id else version
```
Comment (patjlm) [inline reconcile/utils/ocm.py:1205]: Reusing `blockedVersions` with a pattern like `{addon_id}/{version-regex}` for addons

**Review by maorfr — APPROVED** [review]
LGTM

File: reconcile/ocm_upgrade_scheduler.py [hunk 2]
```diff
@@ -69,14 +74,22 @@ def fetch_desired_state(
...
         sector_name = upgrade_policy["conditions"].get("sector")
         if sector_name:
             upgrade_policy["conditions"]["sector"] = ocm.sectors[sector_name]
-        desired_state.append(upgrade_policy)
+        if not addons:
+            spec = ocm.clusters[cluster_name].spec
+            upgrade_policy["current_version"] = spec.version
+            upgrade_policy["channel"] = spec.channel
+            desired_state.append(upgrade_policy)
+            continue
```
Comment (geoberle) [inline reconcile/ocm_upgrade_scheduler.py:86]: forgive me that the only thing i have to say is a stupid nit: 
would you consider putting the `addons` code block into an `else` for better readability as you did it in `fetch_current_state`?

**Review by geoberle — APPROVED** [review]

File: reconcile/ocm_upgrade_scheduler.py (hunk 2 above)
Comment (patjlm) [reply reconcile/ocm_upgrade_scheduler.py:86]: Wanted to avoid longer lines :) but I'll do that

Comment (patjlm) [reply reconcile/ocm_upgrade_scheduler.py]: done in [894376f](https://github.com/app-sre/qontract-reconcile/pull/3120/commits/894376fda09fa507556460399447b8ddb089f0c5)

**Pull request merged**